### DISCUSSION TITLE
How do I integrate kor with HuggingFace Pipeline?

### DISCUSSION THREAD

**Discussion by Seshank-k** [eyurtsev/kor#186] [General]
To create a HuggingFace Pipeline, you need a PromptTemplate. How do I integrate kor with HuggingFace when I don't really have a prompt template and my schema is defined within kor? Is OpenAI the only way to go? My use case works flawlessly with GPT3.5-turbo, but I'm looking for alternate methods.

**Comment by eyurtsev** [comment]
Hi @Seshank-k sorry for the delay I'm on vacation until the end of the month, so I have limited computer access.

You should be able to use any LLM that's available through lamgchain, for hugging face see if this works:

https://python.langchain.com/docs/modules/model_io/models/llms/integrations/huggingface_hub
